### PR TITLE
Added star method for drawing regular stars

### PR DESCRIPTION
**Pull request by RedShy**
I added 
```python 
pdf.star(self, x, y, rin, rout, corners, rotateDegrees=0, style="FD")
``` 

method with optional degree of rotation by doing minor changes to the great [script](https://github.com/digidigital/Extensions-and-Scripts-for-pyFPDF-fpdf2/tree/main/star) of @digidigital.

I checked the correctness of rotation comparing it with [rotation()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.rotation) method, this in an example:

The above image is obtained with this code
```python
pdf.set_fill_color(255, 0, 0)
pdf.star(20, 50 - deltay, rin, rout, 3, rotateDegrees=0, style="FD")
pdf.star(20, 50, rin, rout, 3, rotateDegrees=90, style="FD")
with pdf.rotation(90, 20, 50 + deltay):
    pdf.star(20, 50 + deltay, rin, rout, 3, rotateDegrees=0, style="FD")

pdf.set_fill_color(0, 255, 255)
pdf.star(50, 50 - deltay, rin, rout, 4, rotateDegrees=0, style="FD")
pdf.star(50, 50, rin, rout, 4, rotateDegrees=45, style="FD")
with pdf.rotation(45, 50, 50 + deltay):
    pdf.star(50, 50 + deltay, rin, rout, 4, rotateDegrees=0, style="FD")

pdf.set_fill_color(255, 255, 0)
pdf.star(80, 50 - deltay, rin, rout, 5, rotateDegrees=0, style="FD")
pdf.star(80, 50, rin, rout, 5, rotateDegrees=135, style="FD")
with pdf.rotation(135, 80, 50 + deltay):
    pdf.star(80, 50 + deltay, rin, rout, 5, rotateDegrees=0, style="FD")

pdf.set_fill_color(255, 0, 255)
pdf.star(110, 50 - deltay, rin, rout, 6, rotateDegrees=0, style="FD")
pdf.star(110, 50, rin, rout, 6, rotateDegrees=270, style="FD")
with pdf.rotation(270, 110, 50 + deltay):
    pdf.star(110, 50 + deltay, rin, rout, 6, rotateDegrees=0, style="FD")
```

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

- [ ] The PR description or comment contains a picture of a cute animal (not mandatory but encouraged 😉)

### CODE REVIEW

File: fpdf/fpdf.py [hunk 1]
```diff
@@ -1369,20 +1369,49 @@ def regular_polygon(self, x, y, numSides, polyWidth, rotateDegrees=0, style=None
...
         points = []
-        i = 1
         for i in range(1, numSides + 1):
             point = centerX + radius * math.cos(
                 math.radians((360 / numSides) * i) + math.radians(rotateDegrees)
             ), centerY + radius * math.sin(
                 math.radians((360 / numSides) * i) + math.radians(rotateDegrees)
             )
             points.append(point)
-            i += 1
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1380]: 👍 nice cleanup :)

File: fpdf/fpdf.py [hunk 2]
```diff
@@ -1369,20 +1369,49 @@ def regular_polygon(self, x, y, numSides, polyWidth, rotateDegrees=0, style=None
...
+    def star(self, x, y, rin, rout, corners, rotateDegrees=0, style=None):
+        """
+        Outputs a regular star with n corners.
+        It can be rotated.
+        It can be drawn (border only), filled (with no border) or both.
+
+        Args:
+            x (float): Abscissa of star's centre.
+            y (float): Ordinate of star's centre.
+            rin (float): radius of internal circle.
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1394]: I wonder if `r_in` / `r_out` would not be slightly clearer, and more conform to Python standard `snake_case` naming of variables. What do you think?

File: fpdf/fpdf.py [hunk 3]
```diff
@@ -1369,20 +1369,49 @@ def regular_polygon(self, x, y, numSides, polyWidth, rotateDegrees=0, style=None
...
+            rotateDegrees (float): Optional degree amount to rotate star clockwise.
+
+            style (fpdf.enums.RenderStyle, str): Optional style of rendering. Possible values are:
+            * `D`: draw border. This is the default value.
+            * `F`: fill.
+            * `DF` or `FD`: draw and fill.
+        """
+        th = math.radians(rotateDegrees)
+        point_list = []
+        for i in range(0, (corners * 2) + 1):
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1406]: Nice, smart & clear code!

File: test/shapes/test_star.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import fpdf
+from test.conftest import assert_pdf_equal
+
+import pytest
+
+
+HERE = Path(__file__).resolve().parent
+
+
+def test_star(tmp_path):
```
Comment (Lucas-C) [inline test/shapes/test_star.py:12]: The tests are great!

When arguments passed to a method are "hardcoded" values and not named variables,
I find it clearer to pass options by name:
```python
df.star(x=15, y=y, r_in=5, r_out=15, rotateDegrees=3, style="D")
```
Making this change here is entirely up to you, I'm OK to merge this file as it is.

**Review by Lucas-C — APPROVED** [review]
Good job overall! Thank you for implementing this.

File: fpdf/fpdf.py (hunk 1 above)
Comment (RedShy) [reply fpdf/fpdf.py:1380]: Trying to apply the boy-scout rule! :)

File: fpdf/fpdf.py (hunk 2 above)
Comment (RedShy) [reply fpdf/fpdf.py:1394]: Absolutely yes! What also about `rotateDegrees `in `rotate_degrees`?

File: test/shapes/test_star.py (hunk 4 above)
Comment (RedShy) [reply test/shapes/test_star.py:12]: Yes, I agree with you, I will keep in mind also for next times

File: fpdf/fpdf.py (hunk 2 above)
Comment (Lucas-C) [reply fpdf/fpdf.py:1394]: Sounds good to me!

File: fpdf/fpdf.py [hunk 5]
```diff
@@ -1369,20 +1369,49 @@ def regular_polygon(self, x, y, numSides, polyWidth, rotateDegrees=0, style=None
...
+        It can be rotated.
+        It can be drawn (border only), filled (with no border) or both.
+
+        Args:
+            x (float): Abscissa of star's centre.
+            y (float): Ordinate of star's centre.
+            r_in (float): radius of internal circle.
+            r_out (float): radius of external circle.
+            corners (int): number of star's corners.
+            rotateDegrees (float): Optional degree amount to rotate star clockwise.
```
Comment (Lucas-C) [inline fpdf/fpdf.py:1397]: The argument name needs to match the new `snake_case`

Comment (RedShy) [reply fpdf/fpdf.py]: Right! I missed it

**Comment by gmischler** [comment]
By far the prettiest PR here in quite a while! :star2:

Apparently, when the `.regular_polygon()` method was added, the creator forgot to mention it in shapes.md. Since you're already busy in that neighbourhood, maybe you could add a section about that one there as well? It will almost be a copy of your section about `.star()`, just with the radii arguments replaced by a single radius.
Mille gracie!

**Comment by Lucas-C** [comment]
Thank you @RedShy, that's awesome!
The shape images are a great addition!

I have **one concern** though, about renaming the `rotateDegrees` parameter of `regular_polygon()` into `rotate_degrees`.
This change will potentially break backward compatibility: existing code that already uses this method & argument won't work anymore after upgrading to the next release of `fpdf2`!

This method is relatively recent as it was introduced in [v2.4.6](https://github.com/PyFPDF/fpdf2/blob/master/CHANGELOG.md#246---2021-11-16) in last november. Still, we should either avoid this renaming **or** properly deprecate the old parameter: this means supporting **both** `rotateDegrees` & `rotate_degrees` for some time + raise a `DeprecationWarning` when `rotateDegrees` is used + mention the deprecation in the docstring + add a mention in `CHANGELOG.md`

It's up to you to go all this way or else just avoid renaming this argument as part of this PR 🙂

**Comment by RedShy** [comment]
Glad that you liked the shape images addition 😊

You are right, I didn't thought about that. Now I understand more why one should be careful to change existing code and be careful when adding new code to a library.

Given that it's relatively recent I would like to go this way. In other cases maybe it was not worth it for just a code style change. 

But it's not only `rotateDegrees `in `rotate_degrees` but also:
`numSides `in `num_sides `
`polyWidth `in `poly_width`. 

Do you think it's worth to change them?

**Comment by Lucas-C** [comment]
> Do you think it's worth to change them?

Honestly I don't know ^^
On one hand it would slightly improve the API coherency, on the other adding deprecated arguments is a bit of a hassle, both for `fpdf2` developpers & users as it "pollutes" a little the code.
Anyway, I'd suggest to do it as part of another PR: isolated "atomic" changes are always better than massive PRs.

**Comment by RedShy** [comment]
Okay then I will revert the changes to the snake convention. I will leave the changes for maybe a future PR.

**Comment by Lucas-C** [comment]
Perfect, merging now!
Thank you for your contribution :)

**Comment by Lucas-C** [comment]
This feature has been included in the new [v2.5.5](https://github.com/PyFPDF/fpdf2/releases/tag/2.5.5) release!

**Pull request merged**